perf_tools/principal_surprise.py

```diff
@@ -0,0 +1,835 @@
+import math
+import pandas
+import numpy
+import dask
+import dask.dataframe
+
+from scipy.stats import lognorm
+from scipy.stats import kstest
+
+from sklearn.decomposition import PCA
+from sklearn.preprocessing import StandardScaler
+from sklearn.impute import SimpleImputer
+
+from scipy.stats import multivariate_normal
+
+from matplotlib import pyplot
+import colorhash
+
+from collections import namedtuple
+
+
+class AbnormalTaskPerfAnalyzer:
```

Comment:
This is fine for first pass but suspect we probably want a second patch renaming this and importing it via __init__.py

Reply:
I think I'd rather see this class in its own file with a file name matching the class name but as snack case